builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -148,6 +153,17 @@ func resourceCloudStackInstanceCreate(d *schema.ResourceData, meta interface{})
 		p.SetUserdata(ud)
 	}
 
+	// If the project_name contains any info, we retreive the project_id
+	if projectName, ok := d.GetOk("project_name"); ok {
+		project, _, err := cs.Project.GetProjectByName(projectName.(string))
```

Comment:
I guess we should extent [this](https://github.com/hashicorp/terraform/blob/master/builtin/providers/cloudstack/resources.go#L30) function so we can use the more generally used pattern:

```go
		projectid, e := retrieveUUID(cs, "project", d.Get("project").(string))
		if e != nil {
			return e.Error()
		}
```

Also this parameter should be a good candidate to accept both a UUID or a name, which will be enabled instantly when using this approach.